arangod/Cluster/FollowerInfo.cpp

```diff
@@ -542,3 +543,29 @@ VPackBuilder FollowerInfo::newShardEntry(VPackSlice oldValue) const {
   }
   return newValue;
 }
+
+uint64_t FollowerInfo::newFollowingTermId(ServerID const& s) noexcept {
+  uint64_t i = 0;
+  // We want the random number to be non-zero:
+  do {
+    i = RandomGenerator::interval(UINT64_MAX);
+  } while (i == 0);
+  try {
+    _followingTermId[s] = i;
+  } catch(std::bad_alloc const&) {
+    i = 1;   // I assume here that I do not get bad_alloc if the key is
+             // already in the map, since it then only has to overwrite
+             // an integer, if the key is not in the map, we default to 1.
+  }
+  return i;
+}
+
+uint64_t FollowerInfo::getFollowingTermId(ServerID const& s) const noexcept {
+  // Note that we assume that find() does not throw!
+  auto it = _followingTermId.find(s);
```

Comment:
I think this function can be called by multiple threads (for multiple followers) concurrently, so `_followingTermId` needs to be access-protected by either a mutex or an RW-lock.

Reply:
Indeed. will fix.

Reply:
Done.